src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -1191,27 +1192,29 @@ bool clang_cpp_convertert::get_function_body(
           const exprt &this_ptr = member.op0(); /* the (this) pointer */
           const struct_union_typet &this_type =
             to_struct_union_type(ns.follow(this_ptr.type().subtype()));
-          bool is_struct = this_type.is_struct();
-          exprt rhs_sym(is_struct ? "struct" : "union", this_type);
-          if (is_struct)
-          {
-            for (const struct_union_typet::componentt &c :
-                 this_type.components())
-              if (c.get_name() == member.component_name())
-                rhs_sym.move_to_operands(rhs);
-              else
-              {
-                member_exprt old(this_ptr, c.get_name(), c.type());
-                rhs_sym.move_to_operands(old);
-              }
-          }
-          else
-          {
-            /* single initializer for unions */
-            rhs_sym.component_name(member.component_name());
-            rhs_sym.move_to_operands(rhs);
-          }
 
+          need_array_init = true;
+
+          symbolt new_symbol;
+
+          new_symbol.name = "array_init$";
+          new_symbol.id = id2string(this_ptr.identifier()) + "_array_init$";
+          new_symbol.type = this_type;
+
+          symbolt &init_sym = *context.move_symbol_to_context(new_symbol);
+
+          init_decl = code_declt(symbol_expr(init_sym));
```

Comment:
I think these new instructions might come from this introduction of a new symbol for each array member. I don't think we need that: symex will take care of renaming the existing symbol to something new. In essence, I think, we just need one new symbol `x` for each constructor call, representing an object of the constructed type - independent of how many array members it has.

The sequence would then be:
```
DECL x // <-- this is the single symbol for this constructor of Base
x.ss = {1} // <-- reuse symbol
*this = x
x.sss = {1} // <-- reuse symbol
*this = x
```

Reply:
That's not it, I just checked. Assignments of array members `x.y = z` are side_effect_exprs of array type, so my earlier change in the C-adjuster converts those to `&(x.y = z)[0]`. Somewhere in the preprocessing is that being pulled apart into two statements `x.y = z` followed by another `&x.y` statement.